### PR TITLE
User and Host name utility functions

### PR DESCRIPTION
**Pull request by white238**
I needed these two functions in Serac and they have a pretty good general purpose.

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @white238

**Comment by white238** [comment]
For some reason these tests don't pass in the docker containers. I'll look into it tomorrow.

**Review by agcapps — APPROVED** [review]
Thanks for this addition, Chris.

File: src/axom/CMakeLists.txt [hunk 1]
```diff
@@ -46,7 +46,7 @@ endif()
 # Combine all component object libraries into a unified library
 blt_add_library(NAME       axom
                 SOURCES    Axom.cpp
-                DEPENDS_ON ${AXOM_COMPONENTS_ENABLED})
+                DEPENDS_ON core ${AXOM_COMPONENTS_ENABLED})
```
Comment (white238) [inline src/axom/CMakeLists.txt:49]: Apparently if you turned off all components (leaving just `core`), `core` was never added to the `axom` combined library.

File: src/axom/core/utilities/System.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#else
+  char infoBuf[LOGIN_NAME_MAX];
+  if(getlogin_r(infoBuf, LOGIN_NAME_MAX) == 0)
+  {
+    userName = std::string(infoBuf);
+  }
+  else
+  {
+    // fallback on getpwuid if getlogin_r fails
+    struct passwd *pwd = getpwuid(getuid());
```
Comment (white238) [inline src/axom/core/utilities/System.cpp:66]: Apparently docker images do not login the same way that "normal" machines do so `getlogin_r` fails.  This falls back on another method that did work in this case.

**Comment by white238** [comment]
@agcapps and @kennyweiss can you glance over this again? I had to do some significant changes to get it to work on docker and osx...

File: src/axom/core/utilities/System.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+{
+#define INFO_BUFFER_SIZE 32767
+
+std::string getHostName()
+{
+  std::string hostName = "";
+#ifdef WIN32
+  TCHAR infoBuf[INFO_BUFFER_SIZE];
+  DWORD bufCharCount = INFO_BUFFER_SIZE;
+  bufCharCount = INFO_BUFFER_SIZE;
```
Comment (agcapps) [inline src/axom/core/utilities/System.cpp:31]: what does line 31 do that didn't get done by line 30?
edit: Also line 52 vs line 51.  If there is a reason for the reassignment, would you please put a brief comment saying this is indeed necessary (and perhaps why it is)?

**Review by rhornung67 — APPROVED** [review]
Looks good to me. Thanks @white238

**Review by agcapps — APPROVED** [review]
Thank you for your service @white238 .  I did have one query about the buffer length variable assigned the same value twice in a row.

File: src/axom/core/utilities/System.cpp (hunk 3 above)
Comment (white238) [reply src/axom/core/utilities/System.cpp:31]: I copy/pasted this chunk from microsoft: https://docs.microsoft.com/en-us/windows/win32/sysinfo/getting-system-information

Comment (white238) [reply src/axom/core/utilities/System.cpp:31]: though that seems maybe its there for clarity? ill remove it.

Comment (rhornung67) [reply src/axom/core/utilities/System.cpp]: microsoft!?!  surely, that will cause problems for users...

**Pull request merged**